### PR TITLE
Support Postgres in SQL CLI and make SQL pluggable for any database 

### PR DESCRIPTION
**Pull request by longquanzheng**
1. Create postgres version of schema, mostly follow this: https://en.wikibooks.org/wiki/Converting_MySQL_to_PostgreSQL

2. Create sql tool for postgres. While doing this, I move mysql in a different module so that both mysql/postgres are plugins for this CLI

### CODE REVIEW

File: tools/sql/conn.go [hunk 1]
```diff
@@ -40,90 +39,99 @@ type (
...
-		`PRIMARY KEY (year, month, update_time));`
+	// Driver is the driver interface that each SQL database needs to implement
+	Driver interface {
+		GetDriverName() string
+		CreateDBConnection(driverName, host string, port int, user string, passwd string, database string) (*sqlx.DB, error)
+		GetReadSchemaVersionSQL() string
+		GetWriteSchemaVersionSQL() string
+		GetWriteSchemaUpdateHistorySQL() string
+		GetCreateSchemaVersionTableSQL() string
+		GetCreateSchemaUpdateHistoryTableSQL() string
```
Comment (venkat1109) [inline tools/sql/conn.go]: [ NOT A BLOCKER ] Don't like the fact that there is an implicit assumption about the parameters of the SQL statement for the caller of this API. I think a ConnectionProvider interface and a ConnectionFactory as opposed to this driver interface would be more flexible & cleaner. However,  I understand it's unlikely that parameters would differ between SQL engines and this is something we can change later when we have a need for flexibility. So, up to you

File: tools/sql-extensions/postgres/driver.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
```
Comment (venkat1109) [inline tools/sql-extensions/postgres/driver.go]: 2019

File: tools/sql-extensions/mysql/driver.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
```
Comment (venkat1109) [inline tools/sql-extensions/mysql/driver.go]: 2019

File: tools/sql/conn.go (hunk 1 above)
Comment (longquanzheng) [reply tools/sql/conn.go]: yeah, let's keep that in mind. It's also easier to manage that we give less flexibility at the beginning. It's also possible that people go to implement https://github.com/uber/cadence/blob/master/tools/common/schema/types.go#L52 this interface which will have most flexibility.

File: schema/postgres/cadence/schema.sql [hunk 4]
```diff
@@ -0,0 +1,255 @@
+CREATE TABLE domains(
+  shard_id INTEGER NOT NULL DEFAULT 54321,
+  id BYTEA NOT NULL,
+  name VARCHAR(255) UNIQUE NOT NULL,
+  --
+	data BYTEA NOT NULL,
+	data_encoding VARCHAR(16) NOT NULL,
+	is_global SMALLINT NOT NULL,
```
Comment (wxing1292) [inline schema/postgres/cadence/schema.sql]: indentation?

File: schema/postgres/cadence/versioned/v0.1/base.sql [hunk 5]
```diff
@@ -0,0 +1,245 @@
+CREATE TABLE domains(
+  shard_id INTEGER NOT NULL DEFAULT 54321,
+  id BYTEA NOT NULL,
+  name VARCHAR(255) UNIQUE NOT NULL,
+  --
+	data BYTEA NOT NULL,
+	data_encoding VARCHAR(16) NOT NULL,
+	is_global SMALLINT NOT NULL,
+  PRIMARY KEY(shard_id, id)
```
Comment (wxing1292) [inline schema/postgres/cadence/versioned/v0.1/base.sql]: indentation plz

File: schema/postgres/cadence/versioned/v0.3/manifest.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (wxing1292) [inline schema/postgres/cadence/versioned/v0.3/manifest.json]: where is v0.2?

**Pull request merged**